src/v/cloud_storage/cache_service.cc

```diff
@@ -512,8 +535,71 @@ ss::future<cache::trim_result> cache::trim_fast(
 }
 
 ss::future<cache::trim_result>
-cache::trim_exhaustive(uint64_t target_size, size_t target_objects) {
+cache::trim_exhaustive(uint64_t size_to_delete, size_t objects_to_delete) {
     trim_result result;
+
+    // Enumerate ALL files in the cache (as opposed to trim_fast that strips out
+    // indices/tx/tmp files)
+    auto [walked_cache_size, _filtered_out, candidates, _]
+      = co_await _walker.walk(_cache_dir.native(), _access_time_tracker);
+
+    // Sort by atime
+    std::sort(candidates.begin(), candidates.end(), [](auto& a, auto& b) {
+        return a.access_time < b.access_time;
+    });
```

Comment:
It's unfortunate we have to do the collection and sorting again here. I haven't thought of a better way that doesn't hurt the fast trim, but still something worth thinking about..

Unrelatedly, perhaps we can avoid stat'ing in the walker based on the entry_path. Not sure it's worth doing, and definitely shouldn't block this PR

Reply:
Yeah, the cache work I've done up to this point has been focussed on making it work properly without rearchitecting, but the overall design needs largely redoing to be really efficient: sharing the trimming work between cores, avoiding an exhaustive enumeration in favor of doing some sampling  to figure out an  approximate timestamp cutoff, and then walking+trimming at the same time, and stopping the walk once we've trimmed enough stuff.

However, the latest product thinking is that the caches/disks are going to be tiny, so in practice the cache may only really keep files around for seconds at a time & not do any long term retention at all -- in that world, there might not be much motivation to make it efficient.